src/rdd/rdd.rs

```diff
@@ -666,6 +667,23 @@ pub trait Rdd: RddBase + 'static {
             other,
         ])?))
     }
+
+    fn zip<S: Data>(
+        &self,
+        second: Arc<dyn Rdd<Item = S>>,
+    ) -> SerArc<dyn Rdd<Item = (Self::Item, S)>>
+        where
+            Self: Clone,
+    {
+        SerArc::new(
+            ZippedPartitionsRdd::<Self::Item, S>::new(
+                Arc::new(
+                    self.clone()
+                ) as Arc<dyn Rdd<Item = Self::Item>>,
+                second.clone()
```

Comment:
this second clone is probably unnecessary

Reply:
> this second clone is probably unnecessary

check it later.